### PR TITLE
Simplify Conway.hs and Main.hs

### PR DESCRIPTION
**Pull request by jdygert**

### CODE REVIEW

File: src/Conway.hs [hunk 1]
```diff
@@ -14,58 +13,53 @@ module Conway
...
+
 basicRule :: Rule
-basicRule g =
-  (alive && numNeighboursAlive `elem` [2, 3]) || (not alive && numNeighboursAlive == 3)
+basicRule g = numNeighboursAlive == 3 || (alive && numNeighboursAlive == 2)
   where
     alive = extract g
-    addCoords (x, y) (x', y') = (x + x', y + y')
-    neighbours = experiment (\s -> addCoords s <$> neighbourCoords) g
+    neighbours = experiment (at neighbourCoords) g
```
Comment (ChrisPenner) [inline src/Conway.hs:49]: Can we do 
```haskell
(neighbourCoords `at`)
```
instead?

File: src/Conway.hs [hunk 2]
```diff
@@ -14,58 +13,53 @@ module Conway
...
-
-instance Distributive VBounded where
+instance Distributive Vector where
   distribute = distributeRep
 
-instance Representable VBounded where
-  type Rep VBounded = Int
-  index (VBounded v) i = v V.! (i `mod` gridSize)
-  tabulate desc = VBounded $ V.generate gridSize desc
+instance Representable Vector where
```
Comment (ChrisPenner) [inline src/Conway.hs:31]: I'd prefer to avoid Orphan Instances, but for such a small project shall we let it slide?

**Review by ChrisPenner — APPROVED** [review]

**Comment by ChrisPenner** [comment]
All great changes, I really like how the main loop cleaned up!

File: src/Conway.hs (hunk 2 above)
Comment (jdygert) [reply src/Conway.hs:31]: I hadn't heard of those before, after some brief googling I can see why. I do think for a small project it's not such a big deal.

File: src/Conway.hs (hunk 1 above)
Comment (jdygert) [reply src/Conway.hs:49]: I was considering that. It probably is more clear, like ```(e `elem`)``` over `(elem e)`.

**Pull request merged**